pkg/operator/reconcile.go

```diff
@@ -68,19 +68,25 @@ func (h *Handler) ReconcileObject(ctx context.Context, object *uns.Unstructured)
 }
 
 // MergeObjectForUpdate prepares a "desired" object to be updated.
-// Some objects, such as Deployments, Services, and Daemonsets require
+// Some objects, such as Deployments and Services require
 // some semantic-aware updates
 func MergeObjectForUpdate(current, updated *uns.Unstructured) error {
 	updated.SetResourceVersion(current.GetResourceVersion())
 
-	// todo: add service, daemonset
 	if err := MergeDeploymentForUpdate(current, updated); err != nil {
 		return err
 	}
 
 	if err := MergeServiceForUpdate(current, updated); err != nil {
 		return err
 	}
+
+	// For all object types, merge annotations.
+	// Run this last, in case any of the more specific merge logic has
+	// changed "updated"
+	mergeAnnotationsExcept(current, updated, nil)
```

Comment:
This looks like it should do the wrong thing for the deployment revision annotation, since you're passing `nil` for `exceptKeys` here. Except that it doesn't because `updated` has already been mutated to have `current`s value, but that's really non-obvious. Anyway, maybe you should just only call mergeAnnotationsExcept here if it wasn't a Deployment or Service